### PR TITLE
Prac 10 feedback

### PR DESCRIPTION
**Pull request by cheyenneIT22**
@Winminoo4444 Hi there i am so sorry for late mention. please review my format i hope it is okay.
Thanks cheyenne.

### CODE REVIEW

File: prac_10/wiki.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+"""CP1404, Prac10, wiki quick start guide:
+https://wikipedia.readthedocs.io/en/latest/quickstart.html"""
+import wikipedia
+from wikipedia import DisambiguationError, PageError
+
+search_phrase = input("Search phrase: ")
```
Comment (Winminoo4444) [inline prac_10/wiki.py:6]: Add input validation to handle edge cases like empty or overly long strings

File: prac_10/wiki.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+while search_phrase != "":
+    # Search using search phrase
+    wikipedia.search(search_phrase)
+    # Print summary and get page
+    try:
+        summary = wikipedia.summary(search_phrase)
```
Comment (Winminoo4444) [inline prac_10/wiki.py:12]: Add an option to customize the number of sentences in the summary

File: prac_10/testing.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    'Hello.'
+    >>> format_sentence('It is an ex parrot.')
+    'It is an ex parrot.'
+    >>> format_sentence('this is a test')
+    'This is a test.'
+    """
+    phrase = phrase.strip()
+    if not phrase.endswith('.'):
+        phrase += '.'
+    return phrase.capitalize()
```
Comment (Winminoo4444) [inline prac_10/testing.py:42]: Consider edge cases like input containing only spaces or punctuation

File: prac_10/testing.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+
+def run_tests():
+    """Run the tests on the functions."""
+    # assert test with no message - used to see if the function works properly
+    assert repeat_string("Python", 1) == "Python"
+    # the test below should now pass
+    assert repeat_string("hi", 2) == "hi hi"
```
Comment (Winminoo4444) [inline prac_10/testing.py:50]: Add more detailed error messages for the assertions to identify issues faster

**Review by Winminoo4444 — COMMENTED** [review]
Good attempt for this practical! Keep a good work brother, and good luck for upcoming exam.
